iOS_SDK/OneSignalSDK/Source/OneSignal.m

```diff
@@ -623,6 +625,8 @@ + (id)initWithLaunchOptions:(NSDictionary*)launchOptions
     if ([OneSignalTrackFirebaseAnalytics libraryExists])
         [OneSignalTrackFirebaseAnalytics init];
     
+    [OneSignalLifecycleObserver registerLifecycleObserver];
```

Comment:
I think we are seeing so many logs of "application/scene didBecomeActive" in the tests since this init gets called for most tests and is duplicating calls.

Reply:
This isn't showing up now in the logs due to the default level is now changed. However if logging is turned up this prints many times per event as it subscribes more than once.

Since this is below the [`initDone` check above](https://github.com/OneSignal/OneSignal-iOS-SDK/pull/714/files#diff-bcb6ac33175109d2d958d0e4391f7aa2R595-R597) this won't be an issue on in a real app.
However thinking we clear the observers in between each test so this is close to a device.

Reply:
This is only an issue with the tests.
It doesn't seem to be breaking any of the test from what I see but it creates a lot of duplicated events since the observer gets registered over and over which creates a lot of log entries if you turn up the logging to VERBOSE

Reply:
👍 I see we have `[OneSignalLifecycleObserver removeObserver];` in the tests now which clean up observers carrying over from test to test.